src/libraries/Common/tests/System/Net/Http/HttpClientHandlerTest.cs

```diff
@@ -2115,6 +2115,47 @@ public async Task SendAsync_MultipleExpected100Responses_ReceivesCorrectResponse
             });
         }
 
+        [Fact]
+        public async Task SendAsync_Expect100Continue_RequestBodyFails_ThrowsContentException()
+        {
+            if (IsWinHttpHandler && UseVersion >= HttpVersion20.Value)
+            {
+                return;
+            }
+            if (!TestAsync && UseVersion >= HttpVersion20.Value)
+            {
+                return;
+            }
+
+            var clientFinished = new TaskCompletionSource<bool>();
+            const string ExpectedMessage = "ThrowingContentException";
+
+            await LoopbackServerFactory.CreateClientAndServerAsync(async uri =>
+            {
+                using (HttpClient client = CreateHttpClient())
+                {
+                    HttpRequestMessage initialMessage = new HttpRequestMessage(HttpMethod.Post, uri) { Version = UseVersion };
+                    initialMessage.Content = new ThrowingContent(() => new Exception(ExpectedMessage));
+                    initialMessage.Headers.ExpectContinue = true;
+                    Exception exception = await Assert.ThrowsAsync<Exception>(() => client.SendAsync(TestAsync, initialMessage));
+                    Assert.Equal(ExpectedMessage, exception.Message);
+
+                    clientFinished.SetResult(true);
+                }
+            }, async server =>
+            {
+                await server.AcceptConnectionAsync(async connection =>
+                {
+                    try
+                    {
+                        await connection.ReadRequestDataAsync(readBody: true);
+                    }
+                    catch { } // Eat errors from client disconnect.
+                    await clientFinished.Task;
```

Comment:
It can hang if there is another bug in the client causing hanging somewhere. Please, consider adding a `TimeoutAfter`.

Reply:
Good idea, changed.